src/Simulation/Core/Submitters/IQirSubmitter.cs

```diff
@@ -30,6 +30,20 @@ public interface IQirSubmitter
         Task<IQuantumMachineJob> SubmitAsync(
             Stream qir, string entryPoint, IReadOnlyList<Argument> arguments, SubmissionOptions options);
 
+        /// <summary>
+        /// Submits a job to execute a QIR program without waiting for execution to complete.
+        /// </summary>
+        /// <param name="qir">The QIR program as a byte stream.</param>
+        /// <param name="entryPoint">The fully-qualified name of the entry point to execute.</param>
+        /// <param name="arguments">List of arguments used to execute the entry point.</param>
+        /// <param name="options">Additional options for the submission.</param>
+        /// <returns>The submitted job.</returns>
+        Task<IQuantumMachineJob> SubmitAsync(
+            Stream qir,
+            string entryPoint,
+            IReadOnlyList<EntryPointArgument> arguments, 
```

Comment:
I'm not sure why there needs to be a separate interface for using the EntryPointArgument data type instead of the Argument data type. What's the difference?

Reply:
That's a good point. Originally, I was thinking on doing it this way because I thought the `Argument` data type was too tightly coupled to Q# data types but it is not and it represents QIR types well. Having said that then I think we don't need this new interface.